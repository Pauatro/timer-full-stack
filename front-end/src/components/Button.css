button{
    background-color: var(--button-primary-color);
    padding: 0.75rem 3rem 0.75rem 3rem;
    font-size: 1rem;
    margin: 0;
    border: none;
    border-radius: 0.5rem;
    color: var(--font-secondary-color)
}

.timer-button-content{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}