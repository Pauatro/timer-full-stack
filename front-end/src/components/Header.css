.header-container{
    height: 8vh;
    width: 100vw;
    padding: 2vh;
    box-sizing: inherit;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}

.theme-selection-container{
    height: 4vh;
    width: 16vh;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}