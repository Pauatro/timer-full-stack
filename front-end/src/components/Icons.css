.header-logo, .sun-icon, .moon-icon{
    height: 4vh;
}

.play-icon, .stop-icon{
    height: 1rem;
    margin-right: 1.5rem;
}