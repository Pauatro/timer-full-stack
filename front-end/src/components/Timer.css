.timer-container{
    height: 92vh;
    width: 100vw;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-family: "Arial";
    color: var(--font-primary-color)
}

.timer-display{
    font-size: 4rem;
    padding-bottom: 2.5rem;
}