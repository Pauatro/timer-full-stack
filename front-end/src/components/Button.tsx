import React from 'react'
import { EThemes } from '../modules/theme/theme.interfaces'
import { transformMsToTimeString } from '../utils/formatting'
import './Button.css'
import { PlayIcon } from './Icons'

interface IButtonProps{
    onClick: ()=>void;
    children?: JSX.Element;
}

const Button = (props: IButtonProps)=>{
    return <button onClick = {props.onClick}>{props.children}</button>
}

interface ITimerButtonProps{
    time: number;
    theme: EThemes;
}

const TimerButton = (props: ITimerButtonProps)=>{

    return (
        <Button onClick = {()=>{}}>
            <div className = "timer-button-content">
                <PlayIcon theme = {props.theme}/>
                <div>
                    {transformMsToTimeString(props.time)}
                </div>
            </div>
        </Button>
    )
}

export {
    Button,
    TimerButton
}