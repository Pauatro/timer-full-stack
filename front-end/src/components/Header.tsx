import React from 'react';
import './Header.css';
import { EThemes } from '../modules/theme/theme.interfaces';
import { HeaderLogo, MoonIcon, SunIcon } from './Icons';
import Switch from './Switch';

const Header = () => {
    return  <header className="header-container">
        <HeaderLogo theme = {EThemes.LIGHT}/>
        <div className = "theme-selection-container">
            <MoonIcon theme = {EThemes.LIGHT}/>
            <Switch name = "header-switch"/>
            <SunIcon theme = {EThemes.LIGHT}/>
        </div>
    </header>
}

export default Header