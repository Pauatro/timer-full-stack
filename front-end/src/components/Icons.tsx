import React from 'react';
import './Icons.css';
import lightLogo from '../resources/logo-light.svg'
import darkLogo from '../resources/logo-dark.svg'
import lightMoonIcon from '../resources/moon-light.svg'
import darkMoonIcon from '../resources/moon-dark.svg'
import lightSunIcon from '../resources/sun-light.svg'
import darkSunIcon from '../resources/sun-dark.svg'
import { EThemes } from '../modules/theme/theme.interfaces'

interface IconProps{
    class: string;
    icon: string;
    alt: string;
}

const Icon = (props: IconProps)=>{
    return (
        <img 
            className = {props.class} 
            src = {props.icon} 
            alt = {props.alt}
        />
    )
}

interface ThemedIconProps{
    theme: EThemes;
}

const HeaderLogos = {
    [EThemes.DARK]: darkLogo,
    [EThemes.LIGHT]: lightLogo
}

const HeaderLogo = (props: ThemedIconProps)=>{
    return (
        <Icon 
            class = "header-logo" 
            icon = {HeaderLogos[props.theme]??lightLogo} 
            alt = "MVST"
        />
    )
}

const SunIcons = {
    [EThemes.DARK]: darkSunIcon,
    [EThemes.LIGHT]: lightSunIcon
}

const SunIcon = (props: ThemedIconProps)=>{
    return (
        <Icon 
            class = "sun-icon" 
            icon = {SunIcons[props.theme]??lightLogo} 
            alt = "Sun"
        />
    )
}

const MoonIcons = {
    [EThemes.DARK]: darkMoonIcon,
    [EThemes.LIGHT]: lightMoonIcon
}

const MoonIcon = (props: ThemedIconProps)=>{
    return (
        <Icon 
            class = "moon-icon" 
            icon = {MoonIcons[props.theme]??lightLogo} 
            alt = "MVST"
        />
    )
}

export {
    HeaderLogo,
    SunIcon,
    MoonIcon
}