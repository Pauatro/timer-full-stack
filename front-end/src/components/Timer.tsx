import React, { useContext } from 'react'
import { transformMsToTimeString } from '../utils/formatting'
import './Timer.css'
import { TimerButton } from './Button'
import { ThemeContext } from '../modules/theme/theme.context'

const Timer = ()=>{

    const { theme } = useContext(ThemeContext)

    return (
        <div className="timer-container">
            <TimeDisplay time = {242361453}/>
            <TimerButton 
                theme = {theme} 
                time = {242361453}
            />
        </div>
    )
}

interface ITimerDisplayProps{
    time: number;
}

const TimeDisplay =(props: ITimerDisplayProps)=>{
    return (
        <div className = "timer-display">
            {transformMsToTimeString(props.time)}
        </div>
    )
}

export default Timer