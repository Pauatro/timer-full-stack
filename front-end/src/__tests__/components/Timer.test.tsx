import React from 'react';
import renderer from 'react-test-renderer';
import { Timer, TimerDisplay } from '../../components/Timer';

it('Timer renders correctly', () => {
  const tree = renderer
    .create(<Timer />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('TimerDisplay renders correctly', () => {
  const tree = renderer
    .create(<TimerDisplay 
      time = {0}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});