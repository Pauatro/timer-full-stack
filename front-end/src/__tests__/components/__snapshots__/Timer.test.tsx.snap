// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer renders correctly 1`] = `
<div
  className="timer-container"
>
  <div
    className="timer-display"
  >
    0:00:00
  </div>
  <button
    onClick={[Function]}
  >
    <div
      className="timer-button-content"
    >
      <img
        alt="Start"
        className="play-icon"
        src="play-icon-light.svg"
      />
      <div>
        0:00:00
      </div>
    </div>
  </button>
</div>
`;

exports[`TimerDisplay renders correctly 1`] = `
<div
  className="timer-display"
>
  0:00:00
</div>
`;
